radix-engine/src/kernel/call_frame.rs

```diff
@@ -1089,34 +1089,39 @@ impl<L: Clone> CallFrame<L> {
         handler: &mut impl CallFrameEventHandler,
         node_id: &NodeId,
     ) -> Result<(), PersistNodeError> {
-        handler
-            .on_persist_node(heap, store, node_id)
-            .map_err(|e| PersistNodeError::NotAllowed(node_id.clone(), e))?;
-
-        let node_substates = match heap.remove_node(node_id) {
-            Ok(substates) => substates,
-            Err(HeapRemoveNodeError::NodeNotFound(node_id)) => {
-                panic!("Frame owned node {:?} not found in heap", node_id)
-            }
-            Err(HeapRemoveNodeError::NodeBorrowed(node_id, count)) => {
-                return Err(PersistNodeError::NodeBorrowed(node_id, count));
-            }
-        };
-        for (_partition_number, module_substates) in &node_substates {
-            for (_substate_key, substate_value) in module_substates {
-                for reference in substate_value.references() {
-                    if !reference.is_global() {
-                        return Err(PersistNodeError::ContainsNonGlobalRef(*reference));
-                    }
+        let mut queue = LinkedList::new();
```

Comment:
Perhaps `VecDeque` would be better?

Reply:
It really depends on the workload, how dynamic the queue is. Hard to tell without serious benchmarks.

`LinkedList` gives "constant time" for each pop/push, but each requires a small malloc each time.

`VecDeque` (ring buffer) is fast in push/pop, if everything fits in the buffer. When resize is needed, it requires large malloc and memcpy, which can be slow.

Reply:
Fair enough, I'd just usually consider using array based lists/queues as a default (especially if size is somewhat bounded) and then using linkedlists as an optimization.